gnucash/gnome-utils/gnc-report-combo.c

```diff
@@ -0,0 +1,595 @@
+                                       N_PROPERTIES,
+                                       report_combo_properties);
+}
+
+/** Initialize a GncReportCombo object.
+ *
+ *  @internal
+ *
+ *  @param grc A pointer to the newly created object.
+ */
+static void
+gnc_report_combo_init (GncReportCombo *grc)
+{
+    GncReportComboPrivate *priv;
+
+    g_return_if_fail (grc != NULL);
+    g_return_if_fail (GNC_IS_REPORT_COMBO(grc));
+
+    priv = GET_PRIVATE(grc);
+
+    // Set the name for this widget so it can be easily manipulated with css
+    gtk_widget_set_name (GTK_WIDGET(grc), "gnc-id-report-combo");
+
+    priv->block_signal = FALSE;
+    priv->active_report_guid = NULL;
+    priv->active_report_name = NULL;
+    priv->popup_shown = FALSE;
+}
+
+/** Dispopse the GncReportCombo object. This function is called from
+ *  the G_Object level to complete the destruction of the object.  It
+ *  should release any memory not previously released by the destroy
+ *  function (i.e. the private data structure), then chain up to the
+ *  parent's destroy function.
+ *
+ *  @param object The object being destroyed.
+ *
+ *  @internal
+ */
+static void
+gnc_report_combo_dispose (GObject *object)
+{
+    /* Do not free the private data structure itself. It is part of
+     * a larger memory block allocated by the type system. */
+
+    G_OBJECT_CLASS (gnc_report_combo_parent_class)->dispose (object);
+}
+
+/** Finalize the GncReportCombo object.  This function is called from
+ *  the G_Object level to complete the destruction of the object.  It
+ *  should release any memory not previously released by the destroy
+ *  function (i.e. the private data structure), then chain up to the
+ *  parent's finalize function.
+ *
+ *  @param object The object being finalized.
+ *
+ *  @internal
+ */
+static void
+gnc_report_combo_finalize (GObject *object)
+{
+    GncReportComboPrivate *priv;
+    GncReportCombo *grc;
+
+    g_return_if_fail (object != NULL);
+    g_return_if_fail (GNC_IS_REPORT_COMBO(object));
+
+    grc = GNC_REPORT_COMBO(object);
+    priv = GET_PRIVATE(grc);
+
+    if (priv->active_report_guid)
+        g_free (priv->active_report_guid);
+
+    if (priv->active_report_name)
+        g_free (priv->active_report_name);
+
+    G_OBJECT_CLASS (gnc_report_combo_parent_class)->finalize (object);
+}
+
+static gboolean
+select_active (GncReportCombo *grc)
+{
+    GncReportComboPrivate *priv = GET_PRIVATE(grc);
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(priv->combo));
+    GtkTreeIter iter;
+    gboolean valid_iter = gtk_tree_model_get_iter_first (model, &iter);
+    gboolean found = FALSE;
+
+    while (valid_iter)
+    {
+        gchar *guid;
+        gtk_tree_model_get (model, &iter, RC_GUID, &guid, -1);
+
+        if (g_strcmp0 (priv->active_report_guid, guid) == 0)
+        {
+            gtk_combo_box_set_active_iter (GTK_COMBO_BOX(priv->combo), &iter);
+            g_free (guid);
+            found = TRUE;
+            break;
```

Comment:
Technically, if you break out of the while loop because the current iter is the one you like, `valid_iter` is still TRUE, so you don't need the separate `found` boolean. You can just continue with the state of `valid_iter` further down.

And more even, if you found what you need, you can just return rather than breaking. The second part of this function will then only be run if there was no active selection yet and hence you can drop the if clause around it in that case and just return false.

Reply:
You may also want to document what the returned boolean means. Otherwise it takes some code deciphering before the reader gets it.
With that I mean, the function name `select_active` suggests an imperative action, but the function has side effects (it may add an entry in the combo list!) and returns a value. Without carefully examining the code and the context in which it's called this is not too obvious. So better add a comment and perhaps improve the function name.

Reply:
Changed...